ARCHITECTURE.md

```diff
@@ -0,0 +1,226 @@
+are intended to support more realistic production deployments of Istio and hitless upgrades with canarying.
+
+## Terminology
+
+Throughout the document, the following terms are used:
+
+- New API: The API directly defined in the
+[IstioControlPlane proto](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/istiocontrolplane_types.proto),
+including feature and component groupings, namespaces and enablement, and per-component K8s settings. The term new API
+in this document excludes the pass-through to Helm values fields in the proto.
+- Old API: The Helm values.yaml API, implicitly defined through the various values.yaml files in the
+[Helm charts](https://github.com/istio/installer) and schematized in the operator through
+[Go structs](https://github.com/istio/operator/blob/master/pkg/apis/istio/v1alpha2/values_types.go).
+
+## The new API
+
+The new API is intended to replace the installation and K8s parts of Helm values.yaml.
+
+### Features and components
+
+The operator has a very similar structure to istio/installer: components are grouped into features.
+The operator API defines functional settings at the feature level. Functional settings are those that performs some
+function in the Istio control plane without necessarily being tied to any one component that runs in a Deployment.
+Component settings are those that necessarily refer to a particular Deployment or Service. For example, the number
+of Pilot replicas is clearly a component setting, because it refers to a component which is a Deployment in the
+cluster. Most K8s platform settings are necessarily component settings.
+The available features and components are as follows:
+
+| Feature | Components |
+|---------|------------|
+Base | CRDs
+Traffic Management | Pilot
+Policy | Policy
+Telemetry | Telemetry
+Security | Citadel
+| | Node agent
+| | Cert manager
+Configuration management | Galley
+Gateways | Ingress gateway
+| | Egress gateway
+AutoInjection | Sidecar injector
+
+Features and components are defined in the
+[name](https://github.com/istio/operator/blob/905dd84e868a0b88c08d95b7ccf14d085d9a6f6b/pkg/name/name.go#L38) package.
+
+### Namespaces
+
+The operator API and underlying Helm charts offer a lot of flexibility in which namespaces features and components are
+installed into. Namespace definitions can be defined and specialized at the global, feature and component level, with
+each lower level overriding the setting of the higher parent level. For example, if the global default namespace
+is defined as:
+
+```yaml
+defaultNamespacePrefix: istio-system
+```
+
+and namespaces are specialized for the security feature and one of the components:
+
+```yaml
+security:
+  components:
+    namespace: istio-security
+    citadel:
+    nodeAgent:
+      namespace: istio-security-nodeagent
+policy:
+  components:
+    policy:
+```
+
+the resulting namespaces will be:
+
+|Component|Namespace|
```

Comment:
md requirees spaces on each side of block in a header please change to `| Component | Namesapce |`

Reply:
Done.